src/app.rs

```diff
@@ -1402,20 +1404,23 @@ impl App {
         }
     }
 
-    async fn ensure_user_is_known(
-        &mut self,
-        uuid: Uuid,
-        profile_key: Vec<u8>,
-        phone_number: PhoneNumber,
-    ) {
+    async fn ensure_user_is_known(&mut self, uuid: Uuid, profile_key: ProfileKey) {
         if !self.try_ensure_user_is_known(uuid, profile_key).await {
-            let phone_number_name = phone_number.format().mode(Mode::E164).to_string();
-            self.data.names.insert(uuid, phone_number_name);
+            let name = self
+                .signal_manager
+                .contact_by_id(uuid)
+                .ok()
+                .flatten()
+                .map_or(uuid.to_string(), |c| match c.address.phonenumber {
```

Comment:
Maybe `map_or_else` is better? It avoid converting uuid to string. Also could you please adjust the doc string of `self.data.names`?

Reply:
You need `uuid.to_string()` at some point but I changed the code to only have it once.